### PR TITLE
Fix #4227: Add specialized versions of j.u.function.Consumer

### LINKED ISSUE #4227: Add j.u.function.{Double|Int|Long|ObjDouble|ObjInt|ObjLong}Consumer
Add remaining specialized j.u.function.Consumer implementations

- [x] java.util.function.BiConsumer
- [x] java.util.function.Consumer
- [ ] java.util.function.DoubleConsumer
- [ ] java.util.function.IntConsumer
- [ ] java.util.function.LongConsumer
- [ ] java.util.function.ObjDoubleConsumer
- [ ] java.util.function.ObjIntConsumer
- [ ] java.util.function.ObjLongConsumer

### PR DESCRIPTION
**Pull request by er1c**
I *think* I got all of the formatting correct this time 🤞

### CODE REVIEW

**Comment by er1c** [comment]
@sjrd or @gzm0 any issues here I need to address?

**Comment by sjrd** [comment]
I haven't found the time to look at this one yet, sorry. I will try and find some time this week.

**Comment by er1c** [comment]
@sjrd no problem!

**Comment by er1c** [comment]
@sjrd ping as a friendly reminder, no rush

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/ObjDoubleConsumerTest.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    val double: ObjDoubleConsumer[JavaList[Double]] = makeConsumer { (t: JavaList[Double], num) =>
+      t.forEach(
+        new Consumer[Double] {
+          def accept(value: Double): Unit = {
+            current.add(value * num)
+            ()
+          }
+        }
+      )
+    }
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/ObjDoubleConsumerTest.scala:37]: That is a fairly complicated test, which involves a number of moving parts that are unrelated to `ObjDoubleConsumer`: `java.util.List`, its `forEach` method, the generic `Consumer[T]`.

Consider a test where the `ObjDoubleConsumer` simply stores its two arguments in an enclosing `var result: (SomeClass, Double)`. There's no need to involve the collections here.

(of course the same comment applies to the other `ObjXConsumerTest`s.

**Review by sjrd — CHANGES_REQUESTED** [review]
Sorry for the long delay.

The implementation looks good. I just find the tests of `ObjXConsumer` too complicated for what they're testing. See below.

**Comment by tarsa** [comment]
I think this PR was completely superseded by https://github.com/scala-js/scala-js/pull/4365

**Comment by sjrd** [comment]
Oh, indeed.

**Pull request closed without merging**